sprint 1

Damir-prg opened the pull request:
Собраны компоненты:

- AppHeader ✅
- BurgerConstructor ✅
- IngredientsDetails ✅
- Modal ✅
- ModalOverlay ✅
- OrderDetails ✅

Добавлены:

- Модальный окна (Открытие/Закрытие/Событие на Esc)

Damir-prg committed:
feat: add AppHeader component

Damir-prg committed:
feat: add PageContainer & ConstructorPage

Damir-prg committed:
feat: add ingredients mock api

Damir-prg committed:
part save: add burder ingredients

Damir-prg committed:
part-save

Damir-prg committed:
part-save

Damir-prg committed:
feat: add context

Damir-prg committed:
feat: add context

Damir-prg committed:
feat: part-save

Damir-prg committed:
feat: fix buns and counter

Damir-prg committed:
-

Commit:
edits

Commit:
add readme

Commit:
feat: add esc event

kyzinatra commented on src/components/BurgerIngredients/index.tsx:
```diff
@@ -0,0 +1,53 @@
+      ingredients?.filter((ingredient) => ingredient.type === ETabs.SAUCE) ||
+      [],
+    [ingredients]
+  );
```
Можно лучше: так как у этих `useMemo()` одинаковые зависимости, то можно не писать 3 раза одно и тоже, а объеденить все в 1

kyzinatra commented on src/components/BurgerIngredients/ui/IngredientCard/IngredientCard.tsx:
```diff
@@ -0,0 +1,90 @@
+        if (!prev) {
+          return [itemWithKey];
+        }
+
```
Надо исправить: круто, что вы уже продумали добавление ингредиентов. Но в следующем спринте вы будете заниматься этой логикой через DnD. На клик по ингредиенту остается функционал открытия модального окна с описанием

kyzinatra commented on src/components/AppHeader/ui/HeaderButton/HeaderButton.tsx:
```diff
@@ -0,0 +1,41 @@
+    <button
+      disabled={disabled}
+      className={classes["header-button"]}
+      onClick={handleClick}>
```
Надо исправить**:** Оберните иконку и текст в ссылку в шапке в ссылку `<a href="#">`, a не кнопку

Например

`<a href="#" class="......">`

…
 `Конструктор` 

….
`</a>`

В следующих спринтах вместо ссылки будет компонент Link из библиотеки react router, а пока сделайте просто ссылкой

kyzinatra commented on src/components/ModalOverlay/index.tsx:
```diff
@@ -0,0 +1,30 @@
+
+  const handleKeyPress = (e: KeyboardEvent) => {
+    if (e.keyCode === 27) {
+      onClose();
```
Можно лучше: `keyCode` устарел. Используйте `.key` или `.code`

kyzinatra commented on src/components/ModalOverlay/index.tsx:
```diff
@@ -0,0 +1,30 @@
+  );
+
+  const handleKeyPress = (e: KeyboardEvent) => {
+    if (e.keyCode === 27) {
```
Можно лучше: магическое число лучше вынести отдельно

kyzinatra commented on src/components/ModalOverlay/index.tsx:
```diff
@@ -0,0 +1,30 @@
+
+    return () => {
+      window.removeEventListener("keydown", handleKeyPress);
+    };
```
Отлично! Круто, что вы очищаете обработчик

kyzinatra commented on src/api/ingredients/index.ts:
```diff
@@ -0,0 +1,14 @@
+  getAll: async (): Promise<TApiResponse<Array<TIngredient>>> => {
+    const response = await fetch(
+      "https://norma.nomoreparties.space/api/ingredients"
+    );
```
Надо исправить: Стоит добавить проверку успешности выполнения с помощью `res.ok` так как `catch` ловит только ошибки в самом запросе, но если сервер вернет к примеру 301, то ее `catch` не поймает.

[Подробнее:](https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch) (https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch)

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. В целом вся верстка сделана хорошо. Стоит только ее синхронизировать с заданием спринта, потому что дальше на ее основе вы будете делать много функционала поверх ^-^
Подробнее: 

**Что важно исправить:**

-  круто, что вы уже продумали добавление ингредиентов. Но в следующем спринте вы будете заниматься этой логикой через DnD. На клик по ингредиенту остается функционал открытия модального окна с описанием
- Стоит добавить проверку `res.ok`
- ** Оберните иконку и текст в ссылку в шапке в ссылку `<a href="#">`, a не кнопку

Например

`<a href="#" class="......">`

…
 `Конструктор` 

….
`</a>`

В следующих спринтах вместо ссылки будет компонент Link из библиотеки react router, а пока сделайте просто ссылкой


**Можно лучше:**

-  так как у этих `useMemo()` одинаковые зависимости, то можно не писать 3 раза одно и тоже, а объеденить все в 1
-  `keyCode` устарел. Используйте `.key` или `.code`
-  магическое число лучше вынести отдельно

**Что понравилось:**

-  Круто, что вы очищаете обработчик

**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

Damir-prg committed:
refactor: Правки с ревью

BurgerIngredients: Объеденил мемоизированные элементы
IngredientCard: изменил событие по клику с добавлением в заказ на откртие модального окна с "деталями"
HeaderButton: Изменил тег "button" на "a" с href="#"
ModalOverlay: Изменил устаревший keyCode на key
ingredientsApi: Добавлена проверка response.ok

Damir-prg commented:
BurgerIngredients: Объеденил мемоизированные элементы
IngredientCard: изменил событие по клику с добавлением в заказ на откртие модального окна с "деталями"
HeaderButton: Изменил тег "button" на "a" с href="#"
ModalOverlay: Изменил устаревший keyCode на key
ingredientsApi: Добавлена проверка response.ok
